src/mono/netcore/System.Private.CoreLib/src/System/Reflection/RuntimeModule.cs

```diff
@@ -108,12 +107,11 @@ bool IsResource()
         }
 
         public override
-        Type[] FindTypes(TypeFilter filter, object filterCriteria)
+        Type[] FindTypes(TypeFilter? filter, object? filterCriteria)
         {
             var filtered = new List<Type>();
-            Type[] types = GetTypes();
-            foreach (Type t in types)
-                if (filter(t, filterCriteria))
+            foreach (Type t in GetTypes())
+                if (filter != null && filter(t, filterCriteria))
```

Comment:
This was done to match coreclr and not null ref when a null filter is passed in (which is allowed).